lib/launcher.rb

```diff
@@ -167,10 +191,33 @@ def launch
 
       http_app = http_app(settings.http_port)
       https_app = https_app(settings.https_port)
-      install_webrick_callback!(http_app, https_app)
+      install_http_server_callback!(http_app, https_app)
+
+      http_server_name = @settings.http_server_type
+      https_server_name = @settings.https_server_type
+      if https_app
+        t1 = case https_server_name
+             when :webrick
```

Comment:
I would extract this to a method `add_server()` then then calls `add_webrick_server` or `add_puma_server` to reduce the duplication.

Reply:
executing the server is blocking, so at the moment actually playing around with my threading design.

the need for it, is to support multiple hosts, that the current way of working as a server (puma that is) is blocking.